lib/scope.js

```diff
@@ -237,6 +237,10 @@ AST_Scope.DEFMETHOD("init_scope_vars", function(nesting){
 AST_Lambda.DEFMETHOD("init_scope_vars", function(){
     AST_Scope.prototype.init_scope_vars.apply(this, arguments);
     this.uses_arguments = false;
+
+    var symbol = new AST_VarDef({ name: "arguments" });
```

Comment:
I'll push some tests, given #905 has been pushed to master. I've rebased the patch just before the pr just to take advantage of this.